src/components/Trade/TradeInput.tsx

```diff
@@ -78,6 +78,7 @@ export const TradeInput = () => {
   useSwapperService()
   const [isLoading, setIsLoading] = useState(false)
   const [showQuotes, toggleShowQuotes] = useToggle(false)
+  const [isInitialLoad, setIsInitialLoad] = useState(true)
```

Comment:
This is quite high risk - from the semantics of it, it looks like this is supposed to be true on the first load (mount) of any `<TradeInput />` instance, but this will be set to true on any mount of any `<TradeInput />` component instance (note the double *any*).

I have two concerns with this:

1. The naming, which seem to indicate intent of a loading state for *any* instance, whereas what this actually does is being set to true on the first *mount* of a *specific* instance
2. The `useEffect` below, which will only `setInitialLoad(false)` if `isSwapperApiPending`, which may never be the case during the lifecycle of a specific instance

tl;dr agree with the semantics, but the actual flow isn't matching the intent